presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergUtil.java

```diff
@@ -746,4 +764,120 @@ public static void loadCachingProperties(Map<String, String> properties, Iceberg
         properties.put(IO_MANIFEST_CACHE_MAX_CONTENT_LENGTH, String.valueOf(icebergConfig.getManifestCacheMaxContentLength()));
         properties.put(IO_MANIFEST_CACHE_EXPIRATION_INTERVAL_MS, String.valueOf(icebergConfig.getManifestCacheExpireDuration()));
     }
+
+    public static long getDataSequenceNumber(ContentFile<?> file)
+    {
+        if (file.dataSequenceNumber() != null) {
+            return file.dataSequenceNumber();
+        }
+        return file.fileSequenceNumber();
+    }
+
+    /**
+     * Provides the delete files that need to be applied to the given table snapshot.
+     *
+     * @param table The table to provide deletes for
+     * @param snapshot The snapshot id to use
+     * @param filter Filters to apply during planning
+     * @param requestedPartitionSpec If provided, only delete files for this partition spec will be provided
+     * @param requestedSchema If provided, only delete files with this schema will be provided
+     */
+    public static CloseableIterable<DeleteFile> getDeleteFiles(Table table,
+            long snapshot,
+            TupleDomain<IcebergColumnHandle> filter,
+            Optional<Set<Integer>> requestedPartitionSpec,
+            Optional<Set<Integer>> requestedSchema)
+    {
+        Expression filterExpression = toIcebergExpression(filter);
+        CloseableIterator<FileScanTask> fileTasks = table.newScan().useSnapshot(snapshot).filter(filterExpression).planFiles().iterator();
```

Comment:
Small nit, but just to future proof this code, I'd do one of two things:

1) Add a little check to ensure that the `iterator()` method is called just once.
2) Make this lazy and move the scan planning into the `iterator()` method.

It's unexpected that multiple calls to `iterator()` will not iterate from the beginning.  My preference is on option 2.

Reply:
I changed this line so it doesn't call `.iterator()` and I only call that inside the `iterator()` method. So it should respect the `Iterable` interface now. Does that solution work?